src/ompl/geometric/planners/quotientspace/datastructures/QuotientSpaceGraph.h

```diff
@@ -0,0 +1,241 @@
+    {
+        /// \brief A graph on a quotient-space
+        class QuotientSpaceGraph : public QuotientSpace
+        {
+            typedef QuotientSpace BaseT;
+
+        public:
+            typedef int normalized_index_type;
+
+            /// A configuration in quotient-space
+            class Configuration
+            {
+            public:
+                Configuration() = delete;
+                Configuration(const ompl::base::SpaceInformationPtr &si);
+                Configuration(const ompl::base::SpaceInformationPtr &si, const ompl::base::State *state_);
+                ompl::base::State *state{nullptr};
+                unsigned int total_connection_attempts{0};
+                unsigned int successful_connection_attempts{0};
+                bool on_shortest_path{false};
+
+                /// \brief Element of Probability Density Function (needed to update
+                ///  probability)
+                void *pdf_element;
+                void setPDFElement(void *element_)
+                {
+                    pdf_element = element_;
+                }
+                void *getPDFElement()
+                {
+                    return pdf_element;
+                }
+
+                bool isStart{false};
+                bool isGoal{false};
+
+                /// \brief Index of configuration in boost::graph. Usually in
+                /// the interval [0,num_vertices(graph)], but if vertices are
+                /// deleted or graphs are copied, we sometimes need to map them
+                /// back to [0,num_vertices(graph)] (because otherwise all the
+                /// graph search algorithm cannot find a solution)
+                normalized_index_type index{-1};
+            };
+
+            /// An edge in quotient-space
+            class EdgeInternalState
+            {
+            public:
+                EdgeInternalState() = default;
+                EdgeInternalState(ompl::base::Cost cost_) : cost(cost_){};
+                EdgeInternalState(const EdgeInternalState &eis)
+                {
+                    cost = eis.cost;
+                }
+                void setWeight(double d)
+                {
+                    cost = ompl::base::Cost(d);
+                }
+                ompl::base::Cost getCost()
+                {
+                    return cost;
+                }
+
+            private:
+                ompl::base::Cost cost{+ompl::base::dInf};
+            };
+
+            struct GraphBundle
+            {
+                std::string name{"quotient_graph"};
+            };
+            /// A quotient-graph structure using boost::adjacency_list bundles
+            using Graph = boost::adjacency_list<
+                boost::vecS, 
+                boost::vecS,
+                boost::undirectedS, 
+                Configuration *, 
+                EdgeInternalState,
+                GraphBundle
+            >;
+
+            typedef boost::graph_traits<Graph> BGT;
```

Comment:
OMPL, rather than using `typedef`, uses `using` to create type aliases. Please change all the `typedef` to `using`.